framework/physics/FieldFunction/ff_gridbased_00_constrdestr.cc

```diff
@@ -99,25 +94,91 @@ FieldFunctionGridBased::FieldFunctionGridBased(const std::string& text_name,
                                                double field_value)
   : FieldFunction(text_name, std::move(unknown)),
     sdm_(sdm_ptr),
+    ghosted_field_vector_(MakeFieldVector(*sdm_, UnkManager())),
     local_grid_bounding_box_(sdm_->Grid().GetLocalBoundingBox())
 {
-  const size_t num_local_dofs = sdm_->GetNumLocalDOFs(UnkManager());
-  field_vector_.assign(num_local_dofs, field_value);
+  ghosted_field_vector_->Set(field_value);
+}
 
-  vector_ghost_communicator_ = MakeGhostCommunicator();
+// ##################################################################
+/**Returns a read-only reference to the locally stored field data.*/
+const std::vector<double>& FieldFunctionGridBased::FieldVectorRead() const
+{
+  return ghosted_field_vector_->LocalSTLData();
+}
+/**Returns a reference to the locally stored field data.*/
+std::vector<double>& FieldFunctionGridBased::FieldVector()
+{
+  return ghosted_field_vector_->LocalSTLData();
 }
 
 // ##################################################################
-/**Private method for creating the ghost communicator.*/
-std::shared_ptr<chi_math::VectorGhostCommunicator>
-chi_physics::FieldFunctionGridBased::MakeGhostCommunicator()
+/**Private method for creating the spatial discretization method.*/
+chi_math::SMDPtr
+FieldFunctionGridBased::MakeSDM(const chi::InputParameters& params)
+{
+  const auto& user_params = params.ParametersAtAssignment();
+  const auto& grid_ptr = chi_mesh::GetCurrentHandler().GetGrid();
+  const auto sdm_type = params.GetParamValue<std::string>("sdm_type");
```

Comment:
Same general comment as above, "method type" seems redundant. We should look at changing the convention in the future.